src/main/java/fi/fmi/avi/model/swx/amd82/immutable/SpaceWeatherAdvisoryAmd82Impl.java

```diff
@@ -117,16 +120,19 @@ public static Builder fromAmd79(final SpaceWeatherAdvisoryAmd79 value) {
                     Builder::setTranslationCentreName, //
                     Builder::setTranslationTime, //
                     Builder::setTranslatedTAC);
-            return builder//
+            builder//
                     .setIssuingCenter(IssuingCenterImpl.Builder.fromAmd79(value.getIssuingCenter()).build())
                     .setAdvisoryNumber(AdvisoryNumberImpl.Builder.fromAmd79(value.getAdvisoryNumber()).build())
-                    .setReplaceAdvisoryNumber(value.getReplaceAdvisoryNumber().map(advisoryNumber ->
-                            AdvisoryNumberImpl.Builder.fromAmd79(advisoryNumber).build()))
                     .addAllPhenomena(value.getPhenomena().stream()
                             .map(phenomenon -> SpaceWeatherPhenomenon.valueOf(phenomenon.name())))
                     .addAllAnalyses(value.getAnalyses().stream().map(analysis ->
                             SpaceWeatherAdvisoryAnalysisImpl.Builder.fromAmd79(analysis).build()))
                     .setNextAdvisory(NextAdvisoryImpl.Builder.fromAmd79(value.getNextAdvisory()).build());
+
+            value.getReplaceAdvisoryNumber().ifPresent(replaceAdvisoryNumber ->
+                    builder.addReplaceAdvisoryNumber(AdvisoryNumberImpl.Builder.fromAmd79(replaceAdvisoryNumber).build()));
```

Comment:
Just a matter of style, this could be expressed as

```java
                    .addAllReplaceAdvisoryNumber(value.getReplaceAdvisoryNumber()
                            .map(advisoryNumber -> Collections.singletonList(
                                    AdvisoryNumberImpl.Builder.fromAmd79(advisoryNumber).build()))
                            .orElse(Collections.emptyList()))
```

within the chain. Though, your solution is slightly more efficient. Up to you.